components/server/src/workspace/workspace-factory.ts

```diff
@@ -135,15 +136,26 @@ export class WorkspaceFactory {
         const span = TraceContext.startSpan("createForCommit", ctx);
 
         try {
-            const config = await this.configProvider.fetchConfig({span}, user, context);
+            const [ config, project ] = await Promise.all([
+                this.configProvider.fetchConfig({ span }, user, context),
+                this.projectDB.findProjectByCloneUrl(context.repository.cloneUrl),
```

Comment:
Good point, thanks. However, is that still actually a problem when the Project DB has an index on cloneURL?

Reply:
I think so, yes. I cannot tell without measuring how much it contributes to the startup performance, though. But this really is the most important path, that is already way too long. We can keep it like it is now and do an optimization pass later.

Reply:
So, I think we either fetch the project twice (as currently), or we include the entire `ProjectConfig` in the `WorkspaceConfig` shape (and thus store a config multiple times in DB, for every new workspace).

Not sure what the lesser of evils is, or if maybe there is a better way.

Reply:
We can keep it like it is now and do an optimization pass later.

Reply:
I almost had a single project fetch in `configProvider`, with the project config stored in `WorkspaceConfig`, however the code below also needs the project ID. 🤦

So maybe it could be done the other way around, passing the Project to the configProvider, but I 1000% agree about doing optimization passes on workspace start later (soon).

Reply:
I've added a `TODO` comment at this location.